keps/sig-apps/3998-job-success-completion-policy/README.md

```diff
@@ -0,0 +1,604 @@
+  - [ ] (R) Ensure GA e2e tests meet requirements for [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+  - [ ] (R) Minimum Two Week Window for GA e2e tests to prove flake free
+- [x] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+This KEP extends the Job API to allow setting conditions under which an Indexed Job can be declared as succeeded.
+
+## Motivation
+
+There are cases where a batch workload requires an indexed job that want to care 
+only about drivers (leaders) indexes in determining the success or failure of a Job,
+for example MPI and PyTorch etc. This is currently not possible because the indexed job 
+is marked as Completed only if all indexes are succeeded.
+
+Additionally, there are cases where a batch workload can be declared as succeeded if a number of
+pods succeed, regardless of the index. Sometimes the number needs
+because an indexed job author might not know how many indexes the job will have in advance
+when the above batch workload has an autoscale semantic.
+
+Some third-party frameworks have implemented success policy.
+
+- [Kubeflow Training Operator](https://www.kubeflow.org/docs/components/training)
+- [Flux Operator](https://flux-framework.org/flux-operator/)
+
+### Goals
+
+- allow to mark a job as a `Succeeded` even if some of the indexes fail.
+
+### Non-Goals
+
+- Change the existing behavior of `NonIndexed` mode.
+- Change the existing behavior of `Indexed` mode without a SuccessPolicy.
+
+## Proposal
+
+We propose new policies under which a job can be declared as succeeded.
+Those policies can be modeled in the following:
+
+1. An indexed job completes if a set of [x, y, z...] indexes are successful.
+2. An indexed job completes if x or x% of indexes are successful.
+
+### User Stories (Optional)
+
+#### Story 1
+
+As a machine-learning researcher, I run an indexed job which drivers (leaders) are launched
+as index=0,1,2, and workers are launched as index=3+.
+I want to care about only the drivers when the result of job is evaluated.
```

Comment:
In this scenario do all pods have to be Completed or only the index pods?

Reply:
It's only the index pods.
In other words, we want to mark the job as completed/succeded even if indexes excluding indexes, `0, 1, 2` like index `3` fail.